Python/Product/Debugger/Debugger/DebugConnectionListener.cs

```diff
@@ -82,80 +82,83 @@ static class DebugConnectionListener {
+                                if (!received) {
+                                    throw new TimeoutException();
+                                }
+                            } finally {
+                                debugConn.LegacyLocalConnected -= eventReceived;
                             }
-                        } finally {
-                            debugConn.LegacyLocalConnected -= eventReceived;
                         }
-                    }
-
-                    lock (_targets) {
-                        Guid debugGuid;
-                        WeakReference weakProcess;
-                        PythonProcess targetProcess;
-
-                        if (Guid.TryParse(debugId, out debugGuid) &&
-                            _targets.TryGetValue(debugGuid, out weakProcess) &&
-                            (targetProcess = weakProcess.Target as PythonProcess) != null) {
 
-                            if (result == ConnErrorMessages.None) {
-                                targetProcess.Connect(debugConn);
-                                stream = null;
-                                socket = null;
-                                debugConn = null;
+                        lock (_targets) {
+                            Guid debugGuid;
+                            WeakReference weakProcess;
+                            PythonProcess targetProcess;
+
+                            if (Guid.TryParse(debugId, out debugGuid) &&
+                                _targets.TryGetValue(debugGuid, out weakProcess) &&
+                                (targetProcess = weakProcess.Target as PythonProcess) != null) {
+
+                                if (result == ConnErrorMessages.None) {
+                                    targetProcess.Connect(debugConn);
+                                    stream = null;
+                                    socket = null;
+                                    debugConn = null;
+                                } else {
+                                    WriteErrorToOutputWindow(result, targetProcess);
+                                    targetProcess.Unregister();
+                                }
                             } else {
-                                WriteErrorToOutputWindow(result, targetProcess);
-                                targetProcess.Unregister();
+                                Debug.WriteLine("Unknown debug target: {0}", debugId);
                             }
-                        } else {
-                            Debug.WriteLine("Unknown debug target: {0}", debugId);
                         }
+                    } finally {
+                        debugConn?.Dispose();
                     }
+                } catch (IOException) {
+                } catch (SocketException) {
+                } catch (TimeoutException) {
+                } catch (Exception ex) {
+                    Environment.FailFast("DebugConnectionListener.AcceptConnection", ex);
```

Comment:
If I remember correctly, as written originally, it wasn't bringing down the process for me, so it was hard to figure out what went wrong.